### ISSUE TITLE
IntelliJ 2018.1 'modals' are shown as floating windows instead

### ISSUE THREAD

**Issue by raboof** [raboof/notion#62]
For example the file menu and the class search results.

Might be a matter of adding some winprops configuration.

**Comment by moaxcp** [comment]
I just ran into this issue! I'm really not even sure what to do about it so I am switching to netbeans for now. Also do you know when a new version of notion will be released? I noticed there were some changes merged since the last release.

**Comment by raboof** [comment]
Hmm class still seems to be `jetbrains-idea-ce` and instance `sun-awt-X11-XWindowPeer`, so the existing kludge should at least still apply... we're settings [transient_mode](http://notion.sourceforge.net/notionconf/node4.html#1528) to 'current' [in kludges](https://github.com/raboof/notion/blob/master/etc/cfg_kludges.lua#L65), which should mean "The window should be thought of as a transient for the current active client window (if any) even if it is not marked as a transient by the application" but that does not *appear* to be happening.

Perhaps IntellIJ drops being the 'current active client window' before handing over to the popups? More digging needed I suppose...

**Comment by shokre** [comment]
I solved it by disabling IntellJ's internal JRE and using Oracle java8 instead. 
Modal windows, menus, and popups work as they should, and no kludges are needed.

**Comment by shokre** [comment]
Actually Oracle java had some font issues so i switched to OpenJDK and left MerlinRenderingEngine option in vmoptions file.

**Comment by raboof** [comment]
Interesting so you don't get the problem with openjdk either? It looks like the JDK shipping with IDEA idea-IC-181.4203.550 is 1.8.0_152-release, which version did you try?

**Comment by shokre** [comment]
The problem is definitely with included JRE.

### local open-jdk

```
JRE: 1.8.0_162-8u162-b12-0ubuntu0.16.04.2ppa1-b12 amd64
JVM: OpenJDK 64-Bit Server VM by Oracle Corporation
Linux 4.13.0-39-generic
```
Works properly

### local - oracle

```
JRE: 1.8.0_171-b11 amd64
JVM: Java HotSpot(TM) 64-Bit Server VM by Oracle Corporation
Linux 4.13.0-39-generic
```
Font issues

**Comment by shokre** [comment]
Well, each of the Java versions have some issues, for instance:
- Oracle version works better with OpenGL and without MarlinRendering, but still has font issues. 
- OpenJDK works better without OpenGL and with Marlin

So I ended up with OpenJDK, no kludges, and following vmoptions file (which I think is the default that ships with IDE, except for the `-Xms` and `-Xmx` options):
```
-Xms512m
-Xmx2048m
-XX:ReservedCodeCacheSize=240m
-XX:+UseConcMarkSweepGC
-XX:SoftRefLRUPolicyMSPerMB=50
-ea
-Dsun.io.useCanonCaches=false
-Djava.net.preferIPv4Stack=true
-XX:+HeapDumpOnOutOfMemoryError
-XX:-OmitStackTraceInFastThrow
-Dsun.java2d.renderer=sun.java2d.marlin.MarlinRenderingEngine
```

**Comment by shokre** [comment]
Also tried 
```
java 10.0.1 2018-04-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.1+10)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.1+10, mixed mode)
```
and
```
openjdk version "9-internal"
OpenJDK Runtime Environment (build 9-internal+0-2016-04-14-195246.buildd.src)
OpenJDK 64-Bit Server VM (build 9-internal+0-2016-04-14-195246.buildd.src, mixed mode)
```
but both crashed at startup.

**Comment by shokre** [comment]
They build their own JRE:
```
JRE: 1.8.0_152-release-1136-b29 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.13.0-39-generic
```
and probably apply patches to it, why else build and maintain your own JDK?

The thing is that all Java IDE-s ship their own JRE/JDK versions, but from my experience most of the time they work the same or better with the distro provided one.

**Comment by raboof** [comment]
How do you get that output? So far I got:
```
$ ./idea-IC-181.4203.550/jre64/bin/java -version
openjdk version "1.8.0_152-release"
OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b20)
OpenJDK 64-Bit Server VM (build 25.152-b20, mixed mode)
```

**Comment by shokre** [comment]
From the About box in the IDE

**Comment by raboof** [comment]
https://github.com/JetBrains/jdk8u mentions they only make changes to the 'jdk' module, but I haven't found an easy way to view what's different from the upstream jdk yet

**Comment by shokre** [comment]
You could try [comparing branches](https://github.com/JetBrains/jdk8u_jdk/compare/master...openjdk)

But, there is also a possibility that patches come from ubuntu packaging of openjdk.

**Comment by denis-fokin** [comment]
Have you tried this build? 
https://confluence.jetbrains.com/display/IDEADEV/IDEA+2018.1+EAP

Please, share your About IDE information

**Comment by shokre** [comment]
I can confirm that everything works properly without kludges on EAP build using included JRE.

About box info:
```
IntelliJ IDEA 2018.1.3 (Community Edition)
Build #IC-181.4892.27, built on May 3, 2018
JRE: 1.8.0_152-release-1136-b38 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.13.0-39-generic
```

**Comment by moaxcp** [comment]
https://wiki.archlinux.org/index.php/Java_Runtime_Environment_fonts

This may help with font issues in oraclejdk.

**Comment by raboof** [comment]
So we can expect the next IDEA release will work like normal again? Would be great! Let's keep this open until we've verified this is indeed the case.

**Comment by shokre** [comment]
Strange thing, the issue persists if i use JRE from EAP with 2018.1.2

**Comment by shokre** [comment]
I've been using the EAP build for a few days now, and everything works properly with included JRE.
```
IntelliJ IDEA 2018.1.3 (Community Edition)
Build #IC-181.4892.27, built on May 3, 2018
JRE: 1.8.0_152-release-1136-b38 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.13.0-39-generic
```

also tried the final 2018.1.3, and also everything works properly, and no kludges are needed.
```
IntelliJ IDEA 2018.1.3 (Community Edition)
Build #IC-181.4892.42, built on May 8, 2018
JRE: 1.8.0_152-release-1136-b38 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.13.0-39-generic
```

**Comment by raboof** [comment]
Verified this is fixed in the EAP version (idea-IC-181.5087.6)
